cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,125 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient, APITestCase
+
+from cinema.models import Movie, CinemaHall, Genre, Actor
+from cinema.serializers import (
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieImageSerializer,
+    MovieSerializer,
+)
+from cinema.views import MovieViewSet
+
+MOVIE_URL = reverse("cinema:movie-list")
+MOVIE_SESSION_URL = reverse("cinema:moviesession-list")
+
+
+def sample_movie(**params):
+    defaults = {
+        "title": "Sample movie",
+        "description": "Sample description",
+        "duration": 90,
+    }
+    defaults.update(params)
+
+    return Movie.objects.create(**defaults)
+
+
+def sample_genre(**params):
+    defaults = {
+        "name": "Drama",
+    }
+    defaults.update(params)
+
+    return Genre.objects.create(**defaults)
+
+
+def sample_actor(**params):
+    defaults = {"first_name": "George", "last_name": "Clooney"}
+    defaults.update(params)
+
+    return Actor.objects.create(**defaults)
+
+
+def sample_movie_session(**params):
+    cinema_hall = CinemaHall.objects.create(name="Blue", rows=20, seats_in_row=20)
+
+    defaults = {
+        "show_time": "2022-06-02 14:00:00",
+        "movie": None,
+        "cinema_hall": cinema_hall,
+    }
+    defaults.update(params)
+
+
+class MovieTests(TestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test", password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+
+
+class MovieViewSetTests(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test", password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+        self.genre = sample_genre()
+        self.actor = sample_actor()
+        self.movie = sample_movie()
+        self.movie.genres.add(self.genre)
+        self.movie.actors.add(self.actor)
+        self.movie.save()
+
+    def test_list_movies(self):
+        url = (MOVIE_URL)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Code Quality: In your `test_list_movies` method, you are only checking the status code of the response. It would be a good idea to also check the contents of the response to ensure that it contains the expected data. You can do this by comparing the response data to the data of the movie you created in the `setUp` method.